java-checks/src/main/java/org/sonar/java/checks/CloseResourceCheck.java

```diff
@@ -165,494 +63,18 @@ public void visitNode(Tree tree) {
-        executionState = resultingES.parent.overrideBy(resultingES);
-      } else {
-        executionState = resultingES.parent.merge(resultingES);
-      }
-    }
-
-    private boolean isBreakOrReturnStatement(StatementTree statement) {
-      return statement.is(Tree.Kind.BREAK_STATEMENT, Tree.Kind.RETURN_STATEMENT);
-    }
-
-    private boolean switchContainsDefaultLabel(SwitchStatementTree tree) {
-      for (CaseGroupTree caseGroupTree : tree.cases()) {
-        for (CaseLabelTree label : caseGroupTree.labels()) {
-          if ("default".equals(label.caseOrDefaultKeyword().text())) {
-            return true;
-          }
-        }
-      }
-      return false;
-    }
-
-    private boolean lastStatementIsBreakOrReturn(SwitchStatementTree tree) {
-      List<CaseGroupTree> cases = tree.cases();
-      if (!cases.isEmpty()) {
-        List<StatementTree> lastStatements = cases.get(cases.size() - 1).body();
-        return !lastStatements.isEmpty() && isBreakOrReturnStatement(lastStatements.get(lastStatements.size() - 1));
-      }
-      return false;
-    }
-
-    @Override
-    public void visitWhileStatement(WhileStatementTree tree) {
-      scan(tree.condition());
-      visitStatement(tree.statement());
-    }
-
-    @Override
-    public void visitDoWhileStatement(DoWhileStatementTree tree) {
-      visitStatement(tree.statement());
-      scan(tree.condition());
-    }
-
-    @Override
-    public void visitForStatement(ForStatementTree tree) {
-      scan(tree.condition());
-      scan(tree.initializer());
-      scan(tree.update());
-      visitStatement(tree.statement());
-    }
-
-    @Override
-    public void visitForEachStatement(ForEachStatement tree) {
-      scan(tree.variable());
-      scan(tree.expression());
-      visitStatement(tree.statement());
-    }
-
-    private void visitStatement(StatementTree tree) {
-      executionState = new ExecutionState(executionState);
-      scan(tree);
-      executionState = executionState.restoreParent();
-    }
-
-    private Set<Symbol> extractCloseableSymbols(List<VariableTree> variableTrees) {
-      Set<Symbol> symbols = Sets.newHashSet();
-      for (VariableTree variableTree : variableTrees) {
-        Symbol symbol = variableTree.symbol();
-        if (isCloseableOrAutoCloseableSubtype(symbol.type())) {
-          symbols.add(symbol);
+      for (Tree issueTree : visitor.getIssueTrees()) {
+        Type reportedType = null;
+        if(issueTree.is(Tree.Kind.ASSIGNMENT)) {
+          reportedType = ((AssignmentExpressionTree) issueTree).expression().symbolType();
+        } else if(issueTree.is(Tree.Kind.VARIABLE)) {
+          reportedType = ((VariableTree) issueTree).initializer().symbolType();
+        } else if(issueTree.is(Tree.Kind.NEW_CLASS)) {
+          reportedType = ((NewClassTree) issueTree).symbolType();
         }
+        addIssue(issueTree, String.format("Close this \"%s\".", reportedType.name()));
```

Comment:
:new: ![CRITICAL](https://raw.githubusercontent.com/henryju/image-hosting/master/critical.png)Possible null pointer dereference of reportedType in org.sonar.java.checks.CloseResourceCheck.visitNode(Tree)[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ANP_NULL_ON_SOME_PATH)